generated/provider_dependencies.json

```diff
@@ -552,11 +552,8 @@
       "apache-airflow>=2.2.0",
       "pandas>=0.17.1",
       "simple-salesforce>=1.0.0",
-      "tableauserverclient"
     ],
-    "cross-providers-deps": [
-      "tableau"
-    ]
+    "cross-providers-deps": []
```

Comment:
@potiuk can you confirm this is right?
wasn't sure with the recent change of removing provider dependencies from core.
Did I miss something else needed?

Reply:
OK. Got it. This is wrong:

```
    "simple-salesforce>=1.0.0",
    ],
    "cross-providers-deps": []
``` 

Json does not allow comma at the end of the list (after simple-salesforce). This is whey "Build CI" image is failing - when you look at the error raised in build CI you will see this:


> #69 1.592       Exception while loading provider dependencies Expecting value: line 555 column 5 (char 11592)

This is the error from json parser complaining about the coma.

This is an interesting one and a good learning/opportunity to improve this part.

1) You should NOT modify this file by hand. It should be updated automatically by pre-commit. I specifically put it into "generated" folder and added a README there to explain this (because json does not support comments, so you cannot really explain it in the file). But I entirely see (and I expected that) that people will attempt to modify it (this is why I put it in the "generated" folder. to indicate you should not modify it.

2) The pre-commits had no chance to work because of json parsing at image build time (inside setup.py). Otherwise you would likely figure out that you should not modify it, because you would see pre-commit failing while regenerating the file and showing you generated diff in the CI. This did not happen - because CI image failed to build because of json parsing error. This is a bit chicken-egg problem if someone modifies the .json in the way that fails parsing.

Proposals how to improve (besides fixing it):

* I can simply parse the .json outside of the setup.py as part of the CI job building the image. Then I could provide a very explicit message "PLEASE DO NOT modify this file".
* I can also run only that `pre-commit` before building the image to regenerate it. This might be nicer because it will not stop the image from building (the regenerated .json file will be ALWAYS good because it is well, generated. Then tthe whole build and test process will continue - but static checks will fail when running the same pre-commit in the "CI" workflow.  - > I think this is best way

Does any of you have any better idea how to signal users they should not modify this file :) @ashb @kaxil @eladkal :). I thought I went a bit above and beyond but even that did not prevent a committer and PMC member to THINK they should modify it :). I am not sure if I have any more ideas :P

Reply:
@potiuk I modified this file only because pre commit asked me too I didnt even know about this file before the pre commit error

Reply:
oh wait I read your message wrong.
I think what we have now is OK - I edited it manually because I had issue with pre-commits.
i think we just need to add to the pre-commit message also the sentence "Don't fix manually, run the pre commit to automatically handle this"

Reply:
Yeah. Makes sense

Reply:
https://github.com/apache/airflow/pull/24763